### ISSUE TITLE
error: while setting up the build environment: getting attributes of path '/usr/lib/libSystem.B.dylib': No such file or directory

### ISSUE THREAD

**Issue by winston0410** [NixOS/nix#4951]
**Describe the bug**

I am running Nix on a M1 macbook pro, and it runs alright initially. But after I have installed `xcode` and `Brave browser` and `alacritty` with homebrew, I got the following error.

**Steps To Reproduce**

1. Use `nix-darwin` with the following config

```nix
{ config, pkgs, ... }:

{

  nixpkgs.overlays = [
    (import (builtins.fetchTarball {
      url =
        "https://github.com/nix-community/neovim-nightly-overlay/archive/master.tar.gz";
    }))
  ];

  nixpkgs.config = {
    allowUnfree = true;
    allowUnsupportedSystem = true;
  };

  environment.systemPackages = [
  ];

  # use nix-community cache, run once only
  # cachix use nix-community
  system.stateVersion = 4;

  environment.variables = {
    NIX_REMOTE = "daemon";
    GOPATH = "$HOME/go";
    TMUXP_CONFIGDIR = "$HOME/.tmuxp";
    WEZTERM_CONFIG_FILE = "$HOME/.config/wezterm/config.lua";
    CC = "$(which gcc)";
    CXX = "$(which g++)";
  };

  # shell config
  environment.shells = [ "/run/current-system/sw/bin/zsh" ];
  environment.systemPath = [
    "$HOME/go/bin"
    "$HOME/Library/Python/3.9/bin"
    "$HOME/.npm-global/bin"
    "$HOME/.self-built/bin"
    "$HOME/.local/bin"
  ];

  environment.shellAliases.find = "fd";
  environment.shellAliases.grep = "rg";
  environment.shellAliases.vi = "nvim";
  environment.shellAliases.vim = "nvim";
  environment.shellAliases.vimdiff = "nvim -d";
  environment.shellAliases.ls = "exa --icons";
  environment.shellAliases.gcc-11 = "gcc";
  # environment.shellAliases.cc = "gcc-11";
  # environment.shellAliases."g++" = "g++-11";
  # environment.shellAliases."c++" = "c++-11";
  environment.shellAliases.pip = "pip3.9";
  environment.shellAliases.pfzf =
    "fzf --preview 'bat --style=numbers --color=always --line-range :500 {}'";
  environment.shellAliases.dotfiles =
    "/run/current-system/sw/bin/git --git-dir=$HOME/.dotfiles/ --work-tree=$HOME";
  environment.shellAliases.start = "sh ~/.tmuxp/start.sh";

  environment.interactiveShellInit = ''
    		alias luamake=/Users/hugosum/.config/standalone/lua-language-server/3rd/luamake/luamake
            eval "$(direnv hook zsh)"
          '';

  fonts = {
    enableFontDir = true;
    fonts = with pkgs; [ nerdfonts ];
  };

  # Auto upgrade nix package and the daemon service.
  services.nix-daemon.enable = true;

  programs.zsh = {
    enable = true;
    enableCompletion = true;
    enableBashCompletion = true;
    enableFzfHistory = true;
    enableSyntaxHighlighting = true;
    # promptInit = "prompt off";
    promptInit = "";
  };

  # programs.fzf = {
  # enable = true;
  # enableBashIntegration = true;
  # enableZshIntegration = true;
  # };

  # programs.direnv = {
  # enable = true;
  # enableNixDirenvIntegration = true;
  # # enableBashIntegration = true;
  # enableZshIntegration = true;
  # };

  #programs.ssh.knownHosts = [

  # ]

  system.defaults = {
    NSGlobalDomain = {
      AppleFontSmoothing = 2;
      NSDocumentSaveNewDocumentsToCloud = false;
    };
    finder = {
      QuitMenuItem = true;
      AppleShowAllExtensions = true;
      _FXShowPosixPathInTitle = true;
      FXEnableExtensionChangeWarning = false;
    };
    trackpad = {
      ActuationStrength = 1;
      Clicking = false;
      TrackpadThreeFingerDrag = false;
    };
    dock = {
      autohide = true;
      mru-spaces = false;
      minimize-to-application = true;
    };
  };

  # Keyboard
  system.keyboard = {
    enableKeyMapping = true;
    remapCapsLockToControl = true;
  };

  nix = {
    useDaemon = true;
    binaryCaches = [ "https://cache.nixos.org" ];
    trustedBinaryCaches = [ "https://cache.nixos.org" ];
    buildCores = 0;
    gc.automatic = true;
    useSandbox = false;
    # useSandbox = true;
    # sandboxPaths = [ "/private/tmp" "/private/var/tmp" "/usr/bin/env" ];
  };

}
```

2. Install xcode

3. Install `Brave browser` and `alacritty` through homebrew

```shell
brew install brave-browser
brew install alacritty
```

4. Run `darwin-rebuild switch`

**Expected behavior**

I expect this error should not be seen.

**`nix-env --version` output**

```
nix-env (Nix) 2.3.13
```

**Additional context**

**Comment by domenkozar** [comment]
That's most likely an issue if you enabled sandboxing.

**Comment by winston0410** [comment]
I see. I will try and rollback next time when i see this error

**Comment by domenkozar** [comment]
I've backported fixes for this to 2.3-maintenance branch, when new bugfix Nix release comes out, sandboxing should work again.
